lib/src/Uncertainty/Process/ConditionedGaussianProcess.cxx

```diff
@@ -98,7 +98,47 @@ void ConditionedGaussianProcess::initialize()
   CovarianceMatrix covarianceMatrix(krigingResult_.getConditionalCovariance(vertices));
   // Get the Cholesky factor
   LOGINFO(OSS(false) << "Evaluation of the Cholesky factor");
-  covarianceCholeskyFactor_ = covarianceMatrix.computeCholesky();
+
+  // Boolean flag to tell if the regularization is enough
+  Bool continuationCondition = true;
+  // Scaling factor of the matrix : M-> M + \lambda I with \lambda very small
+  // The regularization is needed for fast decreasing covariance models
+  Scalar maxEV = -1.0;
+  const Scalar startingScaling = ResourceMap::GetAsScalar("GaussianProcess-StartingScaling");
+  const Scalar maximalScaling = ResourceMap::GetAsScalar("GaussianProcess-MaximalScaling");
+  Scalar cumulatedScaling = 0.0;
+  Scalar scaling = startingScaling;
+  while (continuationCondition)
+  {
+    try
+    {
+      covarianceCholeskyFactor_ = covarianceMatrix.computeCholesky();
+      continuationCondition = false;
+    }
+    // If the factorization failed regularize the matrix
+    // Here we use a generic exception as different exceptions may be thrown
+    catch (Exception &)
+    {
+      // If the largest eigenvalue module has not been computed yet...
+      if (maxEV < 0.0)
+      {
+        maxEV = covarianceMatrix.computeLargestEigenValueModule();
+        LOGDEBUG(OSS() << "maxEV=" << maxEV);
+        scaling *= maxEV;
+      }
+      cumulatedScaling += scaling ;
+      LOGDEBUG(OSS() << "scaling=" << scaling << ", cumulatedScaling=" << cumulatedScaling);
+      // Unroll the regularization to optimize the computation
+      for (UnsignedInteger i = 0; i < covarianceMatrix.getDimension(); ++i) covarianceMatrix(i, i) += scaling;
+      scaling *= 2.0;
+      continuationCondition = scaling < maxEV * maximalScaling;
```

Comment:
startingScaling is very small (1e-13) whereas maximalScaling is huge (~1e5)
it is not relevant  your scaling is around 1e5 * maxEV.
Same for GP, GLM 
Is it better to go for a maximumIteration number (like in HMat) ?

Reply:
Looks like 15 should be replaced by 1e-5 in ResourceMap! Good catch! We can also do some arithmetic and say that we do 27 iterations at max. @jschueller the point is to discuss and agree if an approximate method giving a relative error of order 1e-6 is good enough for a sampling method for a process or if we need something else. @sofianehaddad do you prefer to have a starting scaling and a number of iterations (in which case the maximum scaling has to be computed) or a starting scaling and a final scaling (in which case the number of iterations has to be computed)?

IMO this PR is intended to be part of the 1.19 release as it fix a long standing issue, even if the fix is not perfect. A more in-depth rework is needed to simplify the EllipticalDistribution class, in order to have a lightweight Normal class, and use it in any place one want to sample a Gaussian random vector (i.e. GaussianProcess, ConditionalGaussianProcess and some other classes I missed). It should also include an H-mat version, in order to be able to manipulate huge Normal distributions (and also huge Student distributions).

Reply:
I'd go for your first choice. But given time before release, we can for the moment just fix accurately the respective values